cli/names.go

```diff
@@ -0,0 +1,19 @@
+package cli
+
+import (
+	"strings"
+
+	"github.com/docker/cli/cli/names"
+)
+
+var (
+	validContainerNamePattern = names.RestrictedNamePattern
+)
+
+// CheckContainerName check container's name is valid or not
+func CheckContainerName(name string) bool {
+	if len(name) == 0 {
+		return false
+	}
+	return validContainerNamePattern.MatchString(strings.TrimPrefix(name, "/"))
```

Comment:
Why trimming the prefix? Is a container "/aaaa" valid? @thaJeztah WDYT?

Reply:
Yes, docker daemon said it's valid.
see https://github.com/moby/moby/blob/master/daemon/names.go#L59

Reply:
Yes, the leading slash is automatically added, and for historic reasons; a container is named `/containera`, and a container _linked_ to that container (using the legacy `--link` option on the default network) gets an internal name `/containera/containerb` (or something along those lines)

Reply:
I don't think it's allowed to manually _create_ a container with a slash though; they should be added automatically by the daemon where needed